src/workerd/io/worker-entrypoint.c++

```diff
@@ -161,6 +161,10 @@ kj::Promise<void> WorkerEntrypoint::request(
        &metrics = incomingRequest->getMetrics(),
        &wrappedResponse = *wrappedResponse, entrypointName = entrypointName]
       (Worker::Lock& lock) mutable {
+    // TODO(cleanup): Here we rely on the root level JS being evaluated synchronously in
```

Comment:
I'd argue you aren't really assuming that, because if it did not run synchronously, that implies it creates some sort of promise, and that promise will capture the async context.

Reply:
By "that promise will capture the async context", I assume you mean that if the implementer doesn't run the JS synchronously, it's their responsibility to ensure that the returned promise captures and propagates the current async context -- and not that the async capture will happen automatically by way of something already built into `kj::Promises` or `jsg`, correct?  Because the global scope's `runAlarm()` seems to currently be doing something along the lines of:
```
kj::Promise<AlarmResult> ServiceWorkerGlobalScope::runAlarm(...) {
  ...
  return context.run([alarmHandler]((Worker::Lock& lock) -> kj::Promise<AlarmResult> {
    return alarmHandler(lock);
  });
}
```
...which I don't think by itself would convey the async context to the inner lambda.  In which case, I suppose this might be considered a latent bug in `runAlarm()`? (but one that has never been important until now)